app/controllers/idv/phone_controller.rb

```diff
@@ -175,6 +176,18 @@ def async_state_done(async_state)
       handle_proofing_failure
     end
 
+    def is_req_from_frontend?
+      return request.headers['HTTP_X_FORM_STEPS_WAIT'] == '1'
+    end
+
+    def is_req_from_verify_step?
+      return request.referer == idv_verify_info_url
+    end
+
+    def should_keep_flash_success?
+      return is_req_from_frontend? && is_req_from_verify_step?
+    end
```

Comment:
style, I would use ruby's implicit return:
```suggestion
    def is_req_from_frontend?
      request.headers['HTTP_X_FORM_STEPS_WAIT'] == '1'
    end

    def is_req_from_verify_step?
      request.referer == idv_verify_info_url
    end

    def should_keep_flash_success?
      is_req_from_frontend? && is_req_from_verify_step?
    end
```